### PR TITLE
ci: Use @swc/jest for some of the packages (no-changelog)

### PR DESCRIPTION
**Pull request by netroy**
We can't use `@swc/jest` for `cli` until `@swc/core` supports emitting decorator metadata, which is needed for packages like typeorm, class-validator, and class-transformer.

### CODE REVIEW

File: packages/cli/test/integration/shared/utils/testServer.ts [hunk 1]
```diff
@@ -177,6 +177,7 @@ export const setupTestServer = ({
 				variables: { controller: variablesController, path: 'variables' },
 			};
 
+			config.set('publicApi.swaggerUi.disabled', true);
```
Comment (netroy) [inline packages/cli/test/integration/shared/utils/testServer.ts:180]: [this was accidentally removed over 3 months ago. ](https://github.com/n8n-io/n8n/pull/6314/files#diff-889fb30882d6aae0146df3148ae6d9fde164e91d53fe8ec7abea02b6a039eb36L23)

File: jest.config.js [hunk 2]
```diff
@@ -1,5 +1,27 @@
...
+		transform: {
+			legacyDecorator: true,
+			decoratorMetadata: true,
+		},
+		target: 'es2020',
+	},
+};
+
+// TODO: remove ts-jest and use swc in `cli` once swc supports emitting decorator metadata
+// https://github.com/swc-project/swc/issues/6824
```
Comment (ivov) [inline jest.config.js:23]: > until `@swc/core` supports emitting decorator metadata

1. `n8n-hosted-backend/packages/dashboard-backend` also has typeorm, class-validator and class-transformer, but [my swc PR for that repo](https://github.com/n8n-io/n8n-hosted-backend/pull/280) works. Based on your explanation it shouldn't work then?
2. TS recently [implemented](https://devblogs.microsoft.com/typescript/announcing-typescript-5-2-beta/#decorator-metadata) the decorator metadata proposal. I expect this means `--emitDecoratorMetadata` is no longer experimental and could be removed. Unfamiliar with the details of decorator metadata, but wondering if this is somehow relevant?

File: jest.config.js [hunk 3]
```diff
@@ -1,5 +1,27 @@
...
+		parser: {
+			syntax: 'typescript',
+			tsx: false,
+			decorators: true,
+		},
+		transform: {
+			legacyDecorator: true,
+			decoratorMetadata: true,
+		},
+		target: 'es2020',
```
Comment (ivov) [inline jest.config.js:18]: Should we take this value from the tsconfig instead of hardcoding it?

File: jest.config.js [hunk 4]
```diff
@@ -1,5 +1,27 @@
...
+const swcJestOptions = {
+	jsc: {
+		parser: {
+			syntax: 'typescript',
+			tsx: false,
+			decorators: true,
+		},
+		transform: {
+			legacyDecorator: true,
+			decoratorMetadata: true,
```
Comment (ivov) [inline jest.config.js:16]: Can you clarify what this option is for? The [docs explanation](https://swc.rs/docs/configuration/compilation#jsctransformdecoratormetadata) would seem to suggest it's to support decorator metadata, but if decorator metadata is unsupported, I'm unclear on what this actually does then.

File: jest.config.js [hunk 5]
```diff
@@ -9,16 +31,14 @@ const tsJestOptions = {
...
-
 /** @type {import('jest').Config} */
 const config = {
 	verbose: true,
 	testEnvironment: 'node',
 	testRegex: '\\.(test|spec)\\.(js|ts)$',
 	testPathIgnorePatterns: ['/dist/', '/node_modules/'],
 	transform: {
-		'^.+\\.ts$': ['ts-jest', tsJestOptions],
+		'^.+\\.ts$': useSwc ? ['@swc/jest', swcJestOptions] : ['ts-jest', tsJestOptions],
```
Comment (ivov) [inline jest.config.js:41]: 1. For some packages we'll be using `tsc` for `build` and `dev` vs. swc for `test`. Curious if there is a risk that the dist outputs might differ, causing us to miss issues?
2. `cli` is where the bulk of BE tests are and the [linked issue](https://github.com/swc-project/swc/issues/6824) says swc is unlikely to support emitting decorator metadata soon. This really limits the upside of this change.

```
core @swc/jest 1.59s
core ts-jest 1.85s

nodes-base @swc/jest 23.978s
nodes-base ts-jest 33.549s

workflow @swc/jest 3.813s
workflow ts-jest 5.33s
```

Wondering if you've tried `esbuild-jest` and found it limited as well? For reference `bun test` doesn't yet support all the Jest APIs we're using, e.g. `jest.restoreAllMocks()`.

**Comment by ivov** [comment]
I notice `Running coverage on untested files...` adds quite some overhead to every test run, but coverage is often (always?) unneeded when running tests locally. We might gain a DX speedup from only enabling coverage in CI.

File: jest.config.js (hunk 2 above)
Comment (netroy) [reply jest.config.js:23]: looks like we had this working in n8n as well in [this PR](https://github.com/n8n-io/n8n/pull/6320/commits/0490f097f8fa90f5f96623d5cbe1b9034b79332f) (that I have no recollection of 🤦🏽 )

File: jest.config.js (hunk 3 above)
Comment (netroy) [reply jest.config.js:18]: we could, but I was assuming that newer targets would mean faster transpilation in tests. but, I actually haven't benchmarked that.

**Comment by netroy** [comment]
The gains here don't seem worth the maintenance overhead of having different testing setups for some packages.
closing this until we can get `swc` or `esbuild` working for `cli` tests 😞

**Pull request closed without merging**